workspaces/cli-server/src/spectacle/index.ts

```diff
@@ -230,7 +230,7 @@ class LocalCliDiff {
     //@TODO: should this be writing output to the file system?
     // Consume stream instantly for now, resulting in a Promise that resolves once exhausted
     this.diffing = AsyncTools.toArray(worker.results);
-    this.diffing.then(() => {
+    worker.onWriteComplete.then(() => {
```

Comment:
Do you think we need to Promise.all this (i.e. `this.diffing` and `worker.onWriteComplete`?). this.diffing is a promise that can be read + resolved. And it seems like the intent is to conver this.diffing into a direct stream to consume from?

Reply:
This whole bit of code can use a once-over from @JaapRood since it's definitely not taking advantage of all the streaming power. onWriteComplete is probably ok to just use fsSink

Reply:
> Do you think we need to Promise.all this (i.e. this.diffing and worker.onWriteComplete?)

It would be the safe thing to do, as you might create a race condition in for the other fork, otherwise.

As @devdoshi pointed out internally, the worker forks a single result stream into two separate streams, allowing them to be piped to the file system, but consumed in-memory as well. In case subsequent actions are dependent on both of those finishing, we'll want to wait for both of them to finish 😅.

> This whole bit of code can use a once-over from @JaapRood since it's definitely not taking advantage of all the streaming power.

That's true, however as long as we're using the file system directly as a sink it might be hard to make tangible improvements. Streaming systems work very robustly when you write results to a single sink, and all consuming then happens from there. This way, you can make consumption completely consistent and very fault tolerant: if it turned up in the sink, it's correct, otherwise it's not. However, file systems don't generally directly allow you to write and read from it at the same time, at least not without some additional mechanisms some people might call databases 😅. 

That said, I'm sure we can clean this up a bit regardless, for at least it could just be less confusing.